### DISCUSSION TITLE
Feature request: machine-local types

### DISCUSSION THREAD

**Discussion by toddlipcon** [p-org/P#679] [Feature Request!]
Currently, it seems like types are always global constructs. Sometimes I want to use a type as an internal implementation detail for a machine's state. It would be nice to be able to define a type nested within a machine scope, in the same way that a function can be defined in that scope.

**Comment by ankushdesai** [comment]
Noted and added to the list of feature requests. We will add it to the language as we see more demand for it.
I am moving it to discussion to keep the voting around.

Thanks!!
